### ISSUE TITLE
refresh images

### ISSUE THREAD

**Issue by pfefferle** [voxpelli/webpage-webmentions#19]
Do you think there is a way to remove (or better refresh) broken images?

Example: http://pfefferle.github.io/openwebicons/contact/

**Comment by voxpelli** [comment]
Yeah, a pain that the URL of the Twitter avatars change :/

I think I talked about that with @snarfed and some others – but not sure if we concluded what one can do about it.

Question is whether this should be considered a Twitter-specific issue or a more generic indieweb issue? And whether the proper solution would be to refetch mention-data every now and then or to cache the original image?

**Comment by snarfed** [comment]
yup, good question! this does come up regularly. i actually switched bridgy to use twitter's "evergreen" profile picture urls a while back, in snarfed/activitystreams-unofficial@dfc3d406a20965a5ed14c9705e3d3c2223c8c3ff (background in https://indiewebcamp.com/Twitter#Profile_Image_URLs), and FB and G+ always used evergreen urls, so you shouldn't need this for bridgy wms for those silos....but it does apply to IG profile pictures. bridgy has a cron job that updates those (snarfed/bridgy#314), but it doesn't resend all past wms for a user when their picture changes.

for indie wms, it's an open question. some discussion in http://indiewebcamp.com/nicknames-cache#Better_Person_Icon_Display . there are interesting subtleties, e.g. when a profile picture changes, do you show it for all past wms from that person, or do you try to store and use each wm's picture from the time it was received? got me.

**Comment by hugoroy** [comment]
Any idea for a workaround solution to avoid the browser displaying the "broken image" icon?
